linera-core/src/client.rs

```diff
@@ -646,16 +645,26 @@ where
             };
             let this = this.clone();
             let local_node = local_node.clone();
-            let stream = stream.filter(move |notification| {
-                Box::pin(Self::process_notification(
-                    this.clone(),
-                    name,
-                    node.clone(),
-                    local_node.clone(),
-                    notification.clone(),
-                ))
+            let (sender, receiver) = mpsc::unbounded();
+            tokio::spawn(async move {
+                while let Some(notification) = stream.next().await {
+                    let accept = Self::process_notification(
+                        this.clone(),
+                        name,
+                        node.clone(),
+                        local_node.clone(),
+                        notification.clone(),
+                    )
+                    .await;
+
+                    if accept {
+                        let Ok(()) = sender.unbounded_send(notification) else {
+                            break;
+                        };
+                    }
```

Comment:
Maybe check `sender.is_closed` in an else-branch to potentially break even if `accept` is false.

Reply:
Oops, thanks! I refactored it to use [`StreamExt::forward`](https://docs.rs/futures/latest/futures/prelude/stream/trait.StreamExt.html#method.forward), so it handles that for us (and makes the code closer to what it was originally).